### ISSUE TITLE
Hide pty host unresponsive notification when it becomes responsive again

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#117990]
Repro:

- Open multiple windows with terminals
- Trigger pty host hang on Windows by launching this task and killing the terminal when it starts:
  ```
		{
			"type": "shell",
			"command": "echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo && echo foo",
			"label": "Test hang"
		},
  ```
- The notification will show up on all windows
- Click restart, the notification will remain on all other windows

cc @eamodio

**Comment by Tyriar** [comment]
Ideally if another heartbeat occurs it should hide the no longer relevant notification.

**Comment by Tyriar** [comment]
May want to consider this for recovery release depending on the number of reports.

**Comment by Tyriar** [comment]
There are 2 parts to this change:

1. Clicking restart pty host should hide the notification on all windows
2. If the pty host becomes responsive again, the disconnect title annotation should go away and the notification should hide on all windows

The second part will be really hard to verify, so here's me faking it out of sources:

![recording (5)](https://user-images.githubusercontent.com/2193314/109991472-34b7f180-7cbf-11eb-991d-15e2eb462865.gif)

**Comment by sanket-bhalerao** [comment]
@Tyriar, for #117542 
I just opened a new empty folder and got this issue again.
![image](https://user-images.githubusercontent.com/9264813/110248537-b5187400-7f97-11eb-8bd0-487b02a3baf3.png)

I have the latest VSCode insiders build installed.
UPDATE: I restarted the VSCode instance in the same folder and the issue did not appear again for the same folder however, I am able to reproduce the issue every time I follow the below steps
1. select the "Open Folder" option from the menu 
2. create a new folder in the file chooser and name it something (ex: foo)
3. select the newly created folder and open it.
4. a popup "The connection to the terminal's pty host process is unresponsive, the terminals may stop working." appears every time.
